src/registrar/admin.py

```diff
@@ -362,6 +363,123 @@ class UserContactInline(admin.StackedInline):
     model = models.Contact
 
 
+def analytics(request):
+    """View for the reports page."""
+
+    thirty_days_ago = datetime.datetime.today() - datetime.timedelta(days=30)
+
+    last_30_days_applications = models.DomainApplication.objects.filter(created_at__gt=thirty_days_ago)
+    last_30_days_approved_applications = models.DomainApplication.objects.filter(
+        created_at__gt=thirty_days_ago, status=DomainApplication.ApplicationStatus.APPROVED
+    )
+    avg_approval_time = last_30_days_approved_applications.annotate(
+        approval_time=F("approved_domain__created_at") - F("submission_date")
+    ).aggregate(Avg("approval_time"))["approval_time__avg"]
+    # Format the timedelta to display only days
+
+    if avg_approval_time is not None:
+        avg_approval_time = f"{avg_approval_time.days} days"
+    else:
+        avg_approval_time = "No approvals to use"
+    # The start and end dates are passed as url params
+    start_date = request.GET.get("start_date", "")
+    end_date = request.GET.get("end_date", "")
+
+    start_date_formatted = csv_export.format_start_date(start_date)
+    end_date_formatted = csv_export.format_end_date(end_date)
+
+    filter_managed_domains_start_date = {
+        "domain__permissions__isnull": False,
+        "domain__first_ready__lte": start_date_formatted,
+    }
+    filter_managed_domains_end_date = {
+        "domain__permissions__isnull": False,
+        "domain__first_ready__lte": end_date_formatted,
+    }
```

Comment:
```
filter_managed_domains_start_date = {
        "domain__permissions__isnull": False,
        "domain__first_ready__lte": snaphot_date1_formatted,
        "domain__first_ready__gte": start_date_formatted,
    }
    filter_managed_domains_end_date = {
        "domain__permissions__isnull": False,
        "domain__first_ready__lte": snapshot_date2_formatted,
        "domain__first_ready__gte": start_date_formatted,
    }
```
    
  This would be a refactor that allows the left boundary to be moved on the 2 sets of data